### PR TITLE
Enforce Ghost Depth Constraints on Block Size #109

### PR DESCRIPTION
**Pull request by buketbenek**
Lines 695-729 have been added to Config::read_mesh_ in src/Cello/parameters_Config.cpp file as requested in Issue 109.

### CODE REVIEW

**Comment by buketbenek** [comment]
Is there a typo in the formulation of computing active zone on each block in the issue text?
I have modified the proposed formulation with mesh_root_size[i]/mesh_root_blocks[i] for i=0, 1, 2.
It nows passes the tests locally in our cluster.

**Review by fearmayo — APPROVED** [review]

**Comment by fearmayo** [comment]
Great work @buketbenek . @mabruzzo could you have a look at these changes and see if they work for you?

File: src/Cello/parameters_Config.cpp [hunk 1]
```diff
@@ -692,6 +693,41 @@ void Config::read_mesh_ (Parameters * p) throw()
...
+  // Dimensions of the active zone on each block along each axis
+  int ax = mesh_root_size[0] / mesh_root_blocks[0]; 
+  int ay = mesh_root_size[1] / mesh_root_blocks[1];
+  int az = mesh_root_size[2] / mesh_root_blocks[2];
+
+  //  Constraints on the block size based on the ghost depth
+  if ( mesh_max_level > 0 ) {
+    if ( !(ax >= 2*field_ghost_depth[0] && ay >= 2*field_ghost_depth[1] && az >= 2*field_ghost_depth[2] ) ) {
+      ERROR3 ("Config::read", 
+		"Dimensions of the active zone on each block (%d, %d, %d) should be at least double the size of the ghost depth: ",
```
Comment (mabruzzo) [inline src/Cello/parameters_Config.cpp:715]:
```suggestion
		"Dimensions of the active zone on each block (%d, %d, %d) should be at least double the size of the ghost depth for AMR simulations: ", 
```

File: src/Cello/parameters_Config.cpp [hunk 2]
```diff
@@ -692,6 +693,41 @@ void Config::read_mesh_ (Parameters * p) throw()
...
+  //  Constraints on the block size based on the ghost depth
+  if ( mesh_max_level > 0 ) {
+    if ( !(ax >= 2*field_ghost_depth[0] && ay >= 2*field_ghost_depth[1] && az >= 2*field_ghost_depth[2] ) ) {
+      ERROR3 ("Config::read", 
+		"Dimensions of the active zone on each block (%d, %d, %d) should be at least double the size of the ghost depth: ", 
+		ax, ay, az);
+    }  
+    if ( (ax%2 != 0) || (ay%2 != 0) && (az%2 != 0) ) {
+      ERROR3 ("Config::read",
+  		      "Dimensions of the active zone on each block (%d, %d, %d) should each be even" ,
```
Comment (mabruzzo) [inline src/Cello/parameters_Config.cpp:720]:
```suggestion
  		      "Dimensions of the active zone on each block (%d, %d, %d) should each be even for AMR simulations" ,
```

**Review by mabruzzo — APPROVED** [review]
Sorry for taking so long to look at this. It looks great to me. 

I just had 2 minor suggestions for making the error message slightly more descriptive. But, they're fairly subjective. If you disagree with my suggestions, feel free to resolve my comments without addressing them.

File: src/Cello/parameters_Config.cpp (hunk 2 above)
Comment (buketbenek) [reply src/Cello/parameters_Config.cpp:720]: Good suggestions. Makes the message more clear. Thanks a lot.

**Comment by mabruzzo** [comment]
This has 2 approvals and there doesn't seem to be any conflicts with main. So, I'm going to merge!

**Pull request merged**